pkg/controller/webhook/reconcile.go

```diff
@@ -58,7 +62,47 @@ func (w *Params) ReconcileResources(ctx context.Context, clientset kubernetes.In
 		if _, err := clientset.CoreV1().Secrets(w.Namespace).Update(ctx, webhookServerSecret, metav1.UpdateOptions{}); err != nil {
 			return err
 		}
+		updateOperatorPods(ctx, clientset, w.Namespace)
 	}
 
 	return nil
 }
+
+// updateOperatorPods updates a specific annotation on the pods to speed up secret propagation.
+func updateOperatorPods(ctx context.Context, clientset kubernetes.Interface, operatorNamespace string) {
+	// Get all the pods that are related to control-plane label.
+	labels := metav1.ListOptions{
+		LabelSelector: "control-plane=elastic-operator",
+	}
+	pods, err := clientset.CoreV1().Pods(operatorNamespace).List(ctx, labels)
+	if err != nil {
+		return
+	}
+	for _, pod := range pods.Items {
+		updateOperatorPod(ctx, pod, clientset)
+	}
+}
+
+// updateOperatorPod updates a specific annotation on a single pod to speed up secret propagation.
+func updateOperatorPod(ctx context.Context, pod corev1.Pod, clientset kubernetes.Interface) {
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
Since the [ReconcileResources](https://github.com/elastic/cloud-on-k8s/blob/0d0df39cc1830044a378d79bc3693d60bb52a33b/pkg/controller/webhook/reconcile.go#L28) function is outside the reconciliation loop, I implemented this retry mechanism. Would it be worth it?